README.md

```diff
@@ -152,15 +152,15 @@ Building
 ### Dependencies
 
 Compatible compilers/toolchains:
-* Clang: 12 and later
+* Clang: 14 and later
```

Comment:
This one depends on https://github.com/darktable-org/darktable/issues/15659

Reply:
@kmilos : Indeed, depends on #15659. Given the message saying that Clang 12 could be the cause of an issue I'll certainly only allow for Clang 13.

Reply:
@TurboGit Which reminds me, can you please also fix https://github.com/darktable-org/darktable/commit/2fc3add89f9f7ce74509888af764d05b0128475d#r132715263

Reply:
@kmilos : Done!

Reply:
So please above change 14 to 13. TIA.